### DISCUSSION TITLE
Some questions regarding PDFAssistant, Requirements & LLMs

### DISCUSSION THREAD

**Discussion by robertgro** [agno-agi/agno#175] [General]
Hi,

at first, thank you for your efforts. I really appreciate projects like this in general and your documentation for it's simplicity especially. Still, I got some questions left:

1. How is your PDF Assistant implementation different from other providers respectively what is the advantage compared to an "**_askPDF_**" customGPT4? Am I able to Q&A even bigger pdf documents (>20MB) when I store them in a [Knowledge vector database as suggested](https://docs.phidata.com/knowledge/pdf) beforehand or does it solely depends on the connected models context window length?
2. Any hardware requirements related to your project when I self host it including a local model for example?
3. Do you plan to extend the LLMs supported by your app progressively? (https://docs.phidata.com/llms/introduction)

Kind Regards!

**Comment by ashpreetbedi** [comment]
Thanks for the kind works @robertgro:

> Any hardware requirements related to your project when I self host it including a local model for example?

No requirements for phidata, just whatever models you run. For example if you have 16gb RAM you'll probably run llama3 8b comfortably.

> Do you plan to extend the LLMs supported by your app progressively?

Yes we plan to support every LLM, if we're missing any please share and we'll add it within a week

> How is your PDF Assistant implementation different from other providers respectively what is the advantage compared to an "askPDF" customGPT4? 

With phidata you control the input, chunks, how its parsed. Everything is open source so you can tune your Assistant to your needs. + you can add custom tools (like making api calls, searching youtube etc). I think it all depends on how much you want to customize your Assistant. We're here to help :)

> Am I able to Q&A even bigger pdf documents (>20MB) when I store them in a [Knowledge vector database as suggested](https://docs.phidata.com/knowledge/pdf) beforehand

Yes you can easily query much larger documents, mainly because you control the vector db -- so add as much as you'd like. With PgVector you can easily go upto 5-10 GB but if you hit 10GB+ the retrieval needs tuning.

> Does it solely depends on the connected models context window length?

@robertgro in this case we first "retrieve relevant chunks" from the PDF and then send to the model for synthesis, so we can QnA over much larger documents even with smaller context window.

Thank you for the questions, let me know if there's anything else i can help with

**Comment by robertgro** [comment]
Thank you for your fast response @ashpreetbedi.

> No requirements for phidata, just whatever models you run. For example if you have 16gb RAM you'll probably run llama3 8b comfortably.

Sounds great! I'm already familiar with running models locally so that I know what my hardware is capable of.

> Yes we plan to support every LLM, if we're missing any please share and we'll add it within a week

In case I'm missing any LLM, I'll report back to you. For now, your support seems to be pretty comprehensive and enough for my needs!

> With phidata you control the input, chunks, how its parsed. Everything is open source so you can tune your Assistant to your needs. + you can add custom tools (like making api calls, searching youtube etc). I think it all depends on how much you want to customize your Assistant. We're here to help :)
> 

Customization is a huge plus in my opinion. And I'll definitely keep an eye on your project.

> Yes you can easily query much larger documents, mainly because you control the vector db -- so add as much as you'd like. With PgVector you can easily go upto 5-10 GB but if you hit 10GB+ the retrieval needs tuning.

> @robertgro in this case we first "retrieve relevant chunks" from the PDF and then send to the model for synthesis, so we can QnA over much larger documents even with smaller context window.
>

Sounds promising. By the time writing, I haven't had a chance to try it out yet, but I'm excited to figure out if Phidata and PgVector are able to deliver reliable results when querying my documents.

> Thank you for the questions, let me know if there's anything else i can help with

Thank you too, I will!
